### ISSUE TITLE
[BUG] Read only user seeing 403 error when viewing Dashboards

### ISSUE THREAD

**Issue by hsiang9431-amzn** [opensearch-project/security-dashboards-plugin#804]
**Describe the bug**
When logging in as a read only user, `403` error with message "Unable to update UI setting" is shown in the browser app.

**To Reproduce**
1. Create a new user
2. Duplicate the role `kibana_user` to `kibana_user_global_ro`
3. Edit `kibana_user_global_ro` so that it has read only permission on `global_tenant`
4. Create index pattern and dashboards in Global tenant
5. Log in as the user, view global tenant and see the error

**Expected behavior**
No error shall be shown

**Plugins**
security frontend and backend

**Screenshots**
<img width="1711" alt="Screen Shot 2021-08-03 at 4 24 30 PM" src="https://user-images.githubusercontent.com/79167042/128098706-4bd363d2-93ca-4957-8396-f4546274db5e.png">

**Comment by hsiang9431-amzn** [comment]
Identified not fixable in security plugin.

Need to transfer to suitable owner

**Comment by zhyuanqi** [comment]
Managed to reproduce this issue and found out the error message was “no permissions for [indices:data/write/update]”. This is expected because you only gives read permission for global tenant but in cluster permission setting part,  both write and read permission are given.  This conflict causes the error message shows up.  If you give both read and write permission to global tenant, error message disappears.

**Comment by hsiang9431-amzn** [comment]
Such behavior is identified being triggered in the function `ensureDefaultIndexPattern`: https://github.com/elastic/kibana/blob/v7.10.2/src/plugins/data/common/index_patterns/index_patterns/ensure_default_index_pattern.ts#L34

Called by an anonymous function: https://github.com/elastic/kibana/blob/v7.10.2/src/plugins/dashboard/public/application/legacy_app.js#L148

**Comment by tmarkley** [comment]
@hsiang9431-amzn does this issue need to be transferred to the https://github.com/opensearch-project/OpenSearch-Dashboards repository?

**Comment by hsiang9431-amzn** [comment]
@tmarkley Yes, please transfer it to https://github.com/opensearch-project/OpenSearch-Dashboards

**Comment by tmarkley** [comment]
@opensearch-project/admin please transfer this to the OpenSearch-Dashboards repository.
